File: 13 - Determining Compund Interest/Python/main.py
P = float(input("What is the principal amount? £"))
r = float(input("What is the rate of interest? "))/100
t = float(input("What is the number of years? "))
n = float(input("What is the number of times the interest is compounded per year? "))

print("£{} invested at {}%% for {} years compounded {} times per year is £{}".format(P, r, t, n, P*(1 + r/n)**(n*t)))

File: 13 - Determining Compund Interest/Haskell/main.hs
main :: IO()
getNum :: String -> IO Float

getNum s = do
    putStrLn("What is the " ++ s ++ "? ")
    n <- readLn :: IO Float
    return n

main = 
    let final principal rate time number = principal * (1 + rate/100 / number) ** (number * time)
    in do
        p <- getNum "principal amount"
        r <- getNum "rate"
        t <- getNum "number of years"
        n <- getNum "number of times the interest is compounded per year"
        putStrLn("£" ++ show p ++ " invested at " ++ show r ++ " for " ++ show t ++ " years compounded " ++ show n ++ " times per year is £" ++ show(final p r t n))